### PR TITLE
Added dllimport/dllexport where necessary

### PR DESCRIPTION
**Pull request by gshimansky**
Added correct dllimport and dllexport attributes necessary to link libraries and executables on windows

### CODE REVIEW

**Review by kurapov-peter — COMMENTED** [review]
Can we have a single header for these definitions? In Shared for instance.

**Comment by gshimansky** [comment]
> Can we have a single header for these definitions? In Shared for instance.

Windows specific is that you need to declare the same function/variable differently depending on whether you compile a DLL that contains this function/variable or you compile a code that links with this DLL.
https://wiki.tcl-lang.org/page/DLLEXPORT+and+DLLIMPORT

- When you compile DLL itself, function/variable has to be declared as `__declspec(dllexport)`. This is optional because we use cmake mode `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` that makes all DLL symbols as dllexport. But we cannot mark them as `dllimport` in the header because this leads to a linker messages about inconsistency.
- When compiling code external to the DLL, all symbols in that DLLs API headers should be marked as `dllimport`. This is mandatory and linker doesn't find those symbols if they are not explicitly marked.

This is the reason for conditional compilation in DLL headers and also different constants for each DLL.

File: omniscidb/Tests/ResultSetBaselineRadixSortTest.cpp [hunk 1]
```diff
@@ -39,7 +39,10 @@ extern std::unique_ptr<CudaMgr_Namespace::CudaMgr> g_cuda_mgr;
 #include <numeric>
 #include <random>
 
-extern bool g_is_test_env;
+#ifdef _WIN32
```
Comment (ienkovich) [inline omniscidb/Tests/ResultSetBaselineRadixSortTest.cpp:42]: Do we have to add this to all extern declarations? Maybe make EXTERN macro then which would work for both Linux and Windows?

**Comment by alexbaden** [comment]
@gshimansky what is the status of this PR?

File: omniscidb/QueryEngine/Execute.cpp [hunk 2]
```diff
@@ -14,8 +14,10 @@
  * limitations under the License.
  */
 
-#include "QueryEngine/Execute.h"
+#include "Shared/funcannotations.h"
+#define QUERYENGINE_EXPORT RUNTIME_EXPORT
```
Comment (ienkovich) [inline omniscidb/QueryEngine/Execute.cpp:18]: I'd add an empty line after such defines to not hide them within a bunch of includes. Otherwise looks good.

**Review by ienkovich — APPROVED** [review]

**Pull request merged**